### PR TITLE
refactor: using immutability wherever it's possible

### PR DESCRIPTION
**Pull request by annibalsilva**
# Description

Modified the code so that we only use mutability if it's needed; most changes were made in DTOs and classes used as REST responses.

Fixes [FSADT-170](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-170)

## Type of change

Refactoring

# How Has This Been Tested?

No functional change was made, so the existent tests were modified to accommodate the new implementation.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

File: src/main/java/ca/bc/gov/backendstartapi/config/CorsConfig.java [hunk 1]
```diff
@@ -23,7 +22,7 @@ public class CorsConfig implements WebMvcConfigurer {
    */
   @Override
   public void addCorsMappings(CorsRegistry registry) {
-    if (!ObjectUtil.isEmptyOrNull(allowedOrigins)) {
+    if (allowedOrigins.length != 0) {
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/backendstartapi/config/CorsConfig.java:25]: I would suggest to keep the older code, as it is null safe

**Review by paulushcgcj — CHANGES_REQUESTED** [review]

**Review by RMCampos — APPROVED** [review]
Great job 🥳

**Review by paulushcgcj — CHANGES_REQUESTED** [review]
The only pending point is that check. The null check that the old method uses is better to prevent possible nullpointer

File: src/main/java/ca/bc/gov/backendstartapi/config/CorsConfig.java (hunk 1 above)
Comment (annibalsilva) [reply src/main/java/ca/bc/gov/backendstartapi/config/CorsConfig.java]: Added a null check to the condition.

**Review by paulushcgcj — APPROVED** [review]

**Pull request merged**